cache/disk/disk.go

```diff
@@ -85,13 +85,55 @@ func New(dir string, maxSizeBytes int64, proxy cache.CacheProxy) *DiskCache {
 	}
 
 	// The eviction callback deletes the file from disk.
+	// This function is only called while the lock is held
+	// by the current goroutine.
 	onEvict := func(key Key, value SizedItem) {
-		// Only remove committed items (as temporary files have a different filename)
+		
+		f := filepath.Join(dir, key.(string))
+
 		if value.(*lruItem).committed {
-			err := os.Remove(filepath.Join(dir, key.(string)))
+			// Common case. Just remove the cache file and we're done.
+			err := os.Remove(f)
 			if err != nil {
-				log.Println(err)
+				log.Printf("ERROR: failed to remove evicted cache file: %s", f)
```

Comment:
Shouldn't this be a fatal error that crashes the server? There seems something seriously wrong if we can't evict?

Reply:
I think fatal errors are OK during startup, but I'm not sure it's a great idea to kill the server so easily during normal operation- especially if it's a transient error. I wonder if there's another reasonable way to notify the admin somehow?